carbon-design-system/carbon-for-ibm-dotcom
# fix(sticky-header): do not consider vertical ToC height in calculations

### Related Ticket(s)

### Description

The StickyHeader utility has logic for how to calculate the total height of elements on a given web page that are "stuck" to the top of the viewport. It exposes a CSS custom property, `--dds-sticky-header-height`, with this calculated value for other page elements to use.

In AEM, we were seeing this value be set much higher than the actual sum height of all stuck elements. It turns out we were adding the height of `<dds-table-of-contents>` elements that are configured to use the vertical presentation, which is unnecessary.

### Changelog

**Changed**

- Removes `<dds-table-of-contents>` in the vertical orientation from `--dds-sticky-header-height` calculation when appropriate.
- Removes `<dds-table-of-contents>` in the horizontal orientation from `--dds-sticky-header-height` calculation until it is sticking to the top of the viewport.
- Fixes incorrectly named properties related to `<dds-leadspace-with-search>`, allowing those elements to be properly factored into calculations.










## Testing Instructions
Visit the [Dotcom Shell deploy preview](https://ibmdotcom-webcomponents.s3.us-east.cloud-object-storage.appdomain.cloud/deploy-previews/10605/index.html?path=/story/components-dotcom-shell--default)'s various Stories and verify the `--dds-sticky-header-height` custom property set on the root element has a value that matches the current "stuck" elements' height.
